examples/rest-api/main.py

```diff
@@ -0,0 +1,354 @@
+        if db_app is None:
+            raise HTTPException(detail=f"App with id {app_id} does not exist, please create it first.", status_code=400)
+
+        app = App.from_config(yaml_path=db_app.config)
+
+        response = app.add(source=body.source, data_type=body.data_type)
+        return DefaultResponse(response=response)
+    except ValueError as ve:
+        if "OPENAI_API_KEY" in str(ve) or "OPENAI_ORGANIZATION" in str(ve):
+            raise HTTPException(
+                detail="OPENAI_API_KEY or OPENAI_ORGANIZATION not set, please set them in your environment variables.",
+                status_code=400,
+            )
+    except Exception as e:
+        raise HTTPException(detail=f"Error occurred: {e}", status_code=400)
+
+
+@app.post(
+    "/{app_id}/query",
+    tags=["Apps"],
+    response_model=DefaultResponse,
+)
+async def query_an_app(body: QueryApp, app_id: str, db: Session = Depends(get_db)):
+    """
+    Query an existing app.\n
+    app_id: The ID of the app. Use "default" for the default app.\n
+    query: The query that you want to ask the App.\n
+    """
+    try:
+        if app_id is None:
+            raise HTTPException(
+                detail="App ID not provided. If you want to use the default app, use 'default' as the app_id.",
+                status_code=400,
+            )
+
+        db_app = get_app(db, app_id)
+
+        if db_app is None:
+            raise HTTPException(detail=f"App with id {app_id} does not exist, please create it first.", status_code=400)
+
+        app = App.from_config(yaml_path=db_app.config)
+
+        response = app.query(body.query)
+        return DefaultResponse(response=response)
+    except ValueError as ve:
+        if "OPENAI_API_KEY" in str(ve) or "OPENAI_ORGANIZATION" in str(ve):
+            raise HTTPException(
+                detail="OPENAI_API_KEY or OPENAI_ORGANIZATION not set, please set them in your environment variables.",
+                status_code=400,
+            )
+    except Exception as e:
+        raise HTTPException(detail=f"Error occurred: {e}", status_code=400)
+
+
+@app.post(
+    "/{app_id}/chat",
+    tags=["Apps"],
+    response_model=DefaultResponse,
+)
+async def chat_with_an_app(body: MessageApp, app_id: str, db: Session = Depends(get_db)):
+    """
+    Query an existing app.\n
+    app_id: The ID of the app. Use "default" for the default app.\n
+    message: The message that you want to send to the App.\n
+    """
+    try:
+        if app_id is None:
+            raise HTTPException(
+                detail="App ID not provided. If you want to use the default app, use 'default' as the app_id.",
+                status_code=400,
+            )
+
+        db_app = get_app(db, app_id)
+
+        if db_app is None:
+            raise HTTPException(detail=f"App with id {app_id} does not exist, please create it first.", status_code=400)
+
+        app = App.from_config(yaml_path=db_app.config)
+
+        response = app.chat(body.message)
```

Comment:
How are you preserving the chat history here? I think you will have to preserve it separately since Embedchain stores history in memory which won't be accessible by subsequent API calls.

Reply:
Ahh true! Do you think we should keep this endpoint then? I guess it is particularly useful in the case of applications where Embedchain's `App` can stay persistent and a separate bot can be created for a specific use case.

Reply:
Please let me know your thoughts on this!

Reply:
Discussed it in a meet, keep it on hold